### PR TITLE
make sure config stays consistent between v1 and v2

### PR DESCRIPTION
**Pull request by xiaohansong**
## What
https://github.com/airbytehq/airbyte-internal-issues/issues/9994

main change:

encryption -> ssl_mode
all ca certificate/password -> renamed
cursor -> replication_method

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceConfigurationJsonObject.kt [hunk 1]
```diff
@@ -96,17 +86,17 @@ class MysqlSourceConfigurationJsonObject : ConfigurationJsonObjectBase() {
     var jdbcUrlParams: String? = null
 
     @JsonIgnore
-    @ConfigurationBuilder(configurationPrefix = "encryption")
-    val encryption = MicronautPropertiesFriendlyEncryption()
+    @ConfigurationBuilder(configurationPrefix = "mode")
+    var encryption = MicronautPropertiesFriendlyEncryption()
```
[the remark this answers is not in the document]
Comment (postamar) [reply airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceConfigurationJsonObject.kt:90]: I'm struggling to remember how this is done, but I don't think it's by making this value nullable. If you want, file a ticket and assign it to me.

**Review by postamar — COMMENTED** [review]
This is good, but this is missing a bunch of unit tests. See source-firetruck for examples. We should at least have a couple test cases where we feed a valid config JSON from legacy source-mysql via the `airbyte.connector.config.json` micronaut properties.

**Comment by xiaohansong** [comment]
Test added!

**Review by postamar — APPROVED** [review]
If the CI is happy then I'm happy

**Pull request merged**